compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -1805,7 +1805,16 @@ public BLangNode transform(TemplateExpressionNode expressionNode) {
         SyntaxKind kind = expressionNode.kind();
         switch (kind) {
             case XML_TEMPLATE_EXPRESSION:
-                return expressionNode.content().get(0).apply(this);
+                SyntaxKind contentKind = expressionNode.content().get(0).kind();
+                switch (contentKind) {
+                    case XML_COMMENT:
+                    case XML_PI:
+                    case XML_ELEMENT:
+                    case XML_EMPTY_ELEMENT:
+                        return expressionNode.content().get(0).apply(this);
```

Comment:
shouldn't we use createExpression() method?

Reply:
We need to use that method, so changed it